toolbox/base/src/RobotInterface.cpp

```diff
@@ -144,9 +145,12 @@ class RobotInterface::impl
 
     bool initializeRemoteControlBoardRemapper()
     {
+        if (!network) {
+            network = std::make_unique<yarp::os::Network>();
+        }
+
         // Initialize the network
-        yarp::os::Network::init();
-        if (!yarp::os::Network::initialized() || !yarp::os::Network::checkNetwork(5.0)) {
+        if (!network->initialized() || !network->checkNetwork(5.0)) {
```

Comment:
nitpick: these methods are static

Reply:
Yes, they are. However analogously to what happens with ROS, storing internally the network object automatically provides RAII. It is also recommended in the [YARP documentation](http://www.yarp.it/classyarp_1_1os_1_1Network.html#ac4eae967d4d518ec56632dfc0c9360f4).

Reply:
I am referring on how you changed the call to `checkNetwork`, `initialized` and `connect`. C++ allows to call a static method on an object, but I've always found it disturbing.

Regarding the RAII I agree, but I am sure there was a reason at the time I used the explicit `init` and `fini`. It might not being needed anymore now, though.

Reply:
I think that @francesco-romano suggests to call them as `yarp::os::Network::initialized()` , so it is more clear that they are actually static methods.

Reply:
Thanks, I got it now. I was thinking if it was necessary to check `pImpl->network != nullptr` and I ended up not doing it, but calling the methods from the singleton is cleaner and it does not have this problem.

Reply:
I don't think the constructor can fail if not for memory error. You can probably assert on it if you want. But I would not be too concerned for that.

Reply:
I thought the same :+1: Anyway, I substituted all the calls of the pimpl member with calls from the singleton. Now the pimpl member is only used to automatically call `init()` and `fini()`.